Python/marshal.c

```diff
@@ -325,6 +325,7 @@ w_ref(PyObject *v, char *flag, WFILE *p)
     if (p->version < 3 || p->hashtable == NULL)
         return 0; /* not writing object references */
 
+    // TODO: is this enough? what if v is a list containing a code object?
```

Comment:
I think it's enough. All we're trying to protect against is values in refs being code objects (because code objects in refs are used as placeholders). Putting the list in refs is fine.